test/e2e/kappcontroller/fetch_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package kappcontroller
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	"github.com/vmware-tanzu/carvel-kapp-controller/test/e2e"
+	"sigs.k8s.io/yaml"
+)
+
+func Test_App_FetchDirname_Single(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+	name := "test-app-fetch-dirname-single"
+
+	appSingleFetchYAML := `---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: test-app-fetch-dirname-single
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+  - inline:
+      paths:
+        file.csv: |
+            Year,Make,Model,Description,Price
+            1997,Ford,E350,"ac, abs, moon",3000.00
+    path: foo
+  template:
+  - ytt:
+      paths:
+      - foo
```

Comment:
instead of using paths=[foo], lets not specify any paths explicitly and use ytt's data.list() to list out paths seen by ytt. then you can have an assertion that expected list is `[foo/file.csv, ...]`